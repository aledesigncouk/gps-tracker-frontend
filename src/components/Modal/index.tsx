import React from "react";
import styles from "@styles/components/Modal.module.scss";

interface ModalProps {
  title: string;
  content: string;
  isOpen: boolean;
  setModal: (value: boolean) => void;
}

const Modal: React.FC<ModalProps> = ({ title, content, isOpen, setModal }) => {
  const closeModal = () => {
    setModal(!isOpen);
  };

  return (
    <>
      {isOpen && (
        <div className={styles.modal}>
          <div className={styles.overlay} onClick={closeModal}>
            <div className={styles.modalContent}>
              <h2>{title}</h2>
              <p>{content}</p>
              <button className={styles.closeModal} onClick={closeModal}>
                Close
              </button>
            </div>
          </div>
        </div>
      )}
    </>
  );
};

export default Modal;
