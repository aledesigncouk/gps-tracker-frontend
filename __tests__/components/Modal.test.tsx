import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Modal from "@components/Modal";

describe("Modal Component", () => {
  let mockSetModal;

  beforeEach(() => {
    mockSetModal = jest.fn();
    jest.clearAllMocks();
  });

  it("renders the modal when isOpen is true", () => {
    render(<Modal title="Test Title" content="Test Content" isOpen={true} setModal={mockSetModal} />);
    expect(screen.getByText("Test Title")).toBeInTheDocument();
    expect(screen.getByText("Test Content")).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /close/i })).toBeInTheDocument();
  });

  it("does not render the modal when isOpen is false", () => {
    render(<Modal title="Test Title" content="Test Content" isOpen={false} setModal={mockSetModal} />);
    expect(screen.queryByText("Test Title")).not.toBeInTheDocument();
    expect(screen.queryByText("Test Content")).not.toBeInTheDocument();
  });

  it("calls setModal when the close button is clicked", () => {
    render(<Modal title="Test Title" content="Test Content" isOpen={true} setModal={mockSetModal} />);
    const closeButton = screen.getByRole("button", { name: /close/i });
    fireEvent.click(closeButton);
    expect(mockSetModal).toHaveBeenCalledWith(false);
  });

  it("calls setModal when the overlay is clicked", () => {
    render(<Modal title="Test Title" content="Test Content" isOpen={true} setModal={mockSetModal} />);
    const overlay = screen.getByText("Test Content").closest("div");
    fireEvent.click(overlay);
    expect(mockSetModal).toHaveBeenCalledWith(false);
  });
});

describe('Modal component ', () => {
  let mockSetModal;
  it('match the snapshot', () => {
    const { asFragment } = render(<Modal title="Test Title" content="Test Content" isOpen={true} setModal={mockSetModal} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

