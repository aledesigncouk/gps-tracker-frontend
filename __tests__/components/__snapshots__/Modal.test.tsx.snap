// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component  match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="overlay"
    >
      <div
        class="modalContent"
      >
        <h2>
          Test Title
        </h2>
        <p>
          Test Content
        </p>
        <button
          class="closeModal"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
